syntax = "proto3";
package kuser;

option cc_generic_services = true; // 生成服务类和rpc方法描述（默认只生成消息类）


message ResultCode // 通用返回结构，所有RPC响应都可以复用ResultCode
{
    int32 errcode = 1; // 错误码，0-成功
    bytes errmsg = 2;  // 错误信息
}

// 登录请求 【用户名，密码】
message LoginRequest
{
    bytes name = 1; 
    bytes pwd = 2;  
}

// 登录响应 【错误码和错误信息，是否登录成功】
message LoginResponse
{
    ResultCode result = 1; 
    bool success = 2;      
}


// 注册请求 【用户id，用户名，密码】
message RegisterRequst
{
    uint32 id = 1;
    bytes name = 2;
    bytes pwd = 3;
}
// 注册响应 【错误码和错误信息，是否注册成功】
message RegisterResponse
{
    ResultCode result = 1;
    bool success = 2;
}


// RPC 服务接口定义，包括两个远程方法  Login  Register
service UserServiceRpc
{
    rpc Login(LoginRequest)         returns(LoginResponse);
    rpc Register(RegisterRequst)    returns(RegisterResponse);
}